syndesis/plugins/org.fusesource.ide.syndesis.extension.core/src/org/fusesource/ide/syndesis/extensions/core/util/SyndesisVersionUtil.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.syndesis.extensions.core.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.model.Model;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.embedder.IMaven;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.fusesource.ide.preferences.StagingRepositoriesUtils;
+import org.fusesource.ide.syndesis.extensions.core.internal.SyndesisExtensionsCoreActivator;
+
+/**
+ * @author lheinema
+ */
+public class SyndesisVersionUtil {
+	
+	public  static final String BOM_GROUPID = "io.syndesis.extension";
+	public  static final String BOM_ARTIFACTID = "extension-bom";
+	public  static final String BOM_TYPE = "pom";
+	public  static final String PROP_SPRINGBOOT_VERSION = "spring-boot.version";
+	public  static final String PROP_CAMEL_VERSION = "camel.version";
+	public  static final String PROP_SYNDESIS_VERSION = "syndesis.version";
+	
+	private static final String FALLBACK_SPRINGBOOT_VERSION = "1.5.8.RELEASE";
+	private static final String FALLBACK_CAMEL_VERSION = "2.20.1";
+	
+	private SyndesisVersionUtil() {
+		// util class
+	}
+	
+	public static boolean isSyndesisVersionExisting(String syndesisVersion, IProgressMonitor monitor) {
+		return checkSyndesisVersionExisting(syndesisVersion, monitor).containsKey(PROP_SYNDESIS_VERSION);
+	}
+	
+	public static Map<String, String> checkSyndesisVersionExisting(String syndesisVersion, IProgressMonitor monitor) {
+		Map<String, String> versions = new HashMap<>();
+		
+		IMaven maven = MavenPlugin.getMaven();
+		List<ArtifactRepository> repos;
+		try {
+			repos = getRepositoryList(maven);
+		} catch (CoreException ex) {
+			SyndesisExtensionsCoreActivator.pluginLog().logError(ex);
+			repos = new ArrayList<>();
+		}
+		
+		try {
+			Artifact bom = maven.resolve(BOM_GROUPID, BOM_ARTIFACTID, syndesisVersion, BOM_TYPE, null, repos, monitor);
+			if (bom != null) {
+				versions.put(PROP_SYNDESIS_VERSION, syndesisVersion);
+				
+				Model bomModel = maven.readModel(bom.getFile());
+				if (bomModel.getProperties().containsKey(PROP_SPRINGBOOT_VERSION)) {
```

Comment:
I'm wondering if bomModel.getProperties() cannot be null. I don't think that we have the case with our Templates but seems a valid use case.